sympy/combinatorics/fp_groups.py

```diff
@@ -688,4 +798,445 @@ def coset_enumeration_c(fp_grp, Y):
+    >>> C232.table
+    [[1, 1, 2, None],
+     [0, 0, None, None],
+     [None, None, None, 0],
+     [None, None, None, None]]
+    >>> C232.table[0][3] = 3; C232.table[3][2] = 0
+    >>> C232.table
+    [[1, 1, 2, 3],
+     [0, 0, None, None],
+     [None, None, None, 0],
+     [None, None, 0, None]]
+    >>> C232.scan(2, y**3)
+    >>> C232.table
+    [[1, 1, 2, 3], [0, 0, None, None], [None, None, 3, 0], [None, None, 0, 2]]
+    >>> C232.table[1][2] = 1; C232.table[1][3] = 1
+    >>> C232.table
+    [[1, 1, 2, 3], [0, 0, 1, 1], [None, None, 3, 0], [None, None, 0, 2]]
+    >>> first_in_class(C232)
+    False
+
+    # TODO:: Sims points out in [Sim94] that performance can be improved by
+    # remembering some of the information computed by ``first_in_class``. If
+    # the ``continue α`` statement is executed at line 14, then the same thing
+    # will happen for that value of α in any descendant of the table C, and so
+    # the values the values of α for which this occurs could profitably be
+    # stored and passed through to the descendants of C. Of course this would
+    # make the code more complicated.
+
+    # The code below is taken directly from the function on page 208 of [Sim94]
+    # ν[α]
+
+    """
+    n = C.n
+    # lamda is the largest numbered point in Ω_c_α which is currently defined
+    lamda = -1
+    # for α ∈ Ω_c, ν[α] is the point in Ω_c_α corresponding to α
+    nu = [None]*n
+    # for α ∈ Ω_c_α, μ[α] is the point in Ω_c corresponding to α
+    mu = [None]*n
+    # mutually ν and μ are the mutually-inverse equivalence maps between
+    # Ω_c_α and Ω_c
+    next_alpha = False
+    # For each 0≠α ∈ [0 .. nc-1], we start by constructing the equivalent
+    # standardized coset table C_α corresponding to H_α
+    for alpha in range(1, n):
+        # reset ν to "None" after previous value of α
+        for beta in range(lamda+1):
+            nu[mu[beta]] = None
+        # try α as the new point 0 in Ω_C_α
+        mu[0] = alpha
+        nu[alpha] = 0
+        # compare corresponding entries in C and C_α
+        lamda = 0
+        for beta in range(n):
+            for x in C.A:
+                gamma = C.table[beta][C.A_dict[x]]
+                delta = C.table[mu[beta]][C.A_dict[x]]
+                # if either of the entries is undefined,
+                # we move with next α
+                if gamma is None or delta is None:
+                    # continue with α
+                    next_alpha = True
+                    break
+                if nu[delta] is None:
+                    # delta becomes the next point in Ω_C_α
+                    lamda += 1
+                    nu[delta] = lamda
+                    mu[lamda] = delta
+                if nu[delta] < gamma:
+                    return False
+                if nu[delta] > gamma:
+                    # continue with α
+                    next_alpha = True
+                    break
+            if next_alpha:
+                break
```

Comment:
Yes, definitely.

On 06/17/2016 03:57 PM, Kalevi Suominen wrote:
>
> In sympy/combinatorics/fp_groups.py 
> <https://github.com/sympy/sympy/pull/11231#discussion_r67490420>:
>
> > +                    # continue with α
> > +                    next_alpha = True
> > +                    break
> > +                if nu[delta] is None:
> > +                    # delta becomes the next point in Ω_C_α
> > +                    lamda += 1
> > +                    nu[delta] = lamda
> > +                    mu[lamda] = delta
> > +                if nu[delta] < gamma:
> > +                    return False
> > +                if nu[delta] > gamma:
> > +                    # continue with α
> > +                    next_alpha = True
> > +                    break
> > +            if next_alpha:
> > +                break
>
> It seems that |next_alpha| should be reset before |break| (|next_alpha 
> = False|).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/sympy/sympy/pull/11231/files/09134253eb3492d3304a8f0ebd223dbecddcdda1#r67490420>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AHWt8Snol98x9yBiE-VxN7bMn3-YSwPqks5qMnaugaJpZM4IzwKC>.
>